src/engine.cc

```diff
@@ -0,0 +1,83 @@
+#include "engine.h"
+#include "gobject.h"
+
+#include <eos-knowledge-content.h>
+
+using namespace v8;
+
+struct CallbackData {
+  Persistent<Promise::Resolver> resolver;
+};
+
+static void EngineGetObjectCallback(GObject *source,
+                                    GAsyncResult *result,
+                                    gpointer user_data)
+{
+  Isolate *isolate = Isolate::GetCurrent ();
+  HandleScope scope(isolate);
+  CallbackData *data = (CallbackData *)user_data;
+
+  GError *error = NULL;
+  EkncContentObjectModel *model;
+
+  Local<Promise::Resolver> resolver = Local<Promise::Resolver>::New(isolate, data->resolver);
+  if ((model = eknc_engine_get_object_finish (EKNC_ENGINE (source), result, &error)))
+    resolver->Resolve(WrapperFromGObject (isolate, G_OBJECT (model)));
+  else
+    resolver->Reject(Exception::TypeError(String::NewFromUtf8(isolate, error->message)));
+
+  /* XXX: For some reason we have to resolve microtasks to the promise wrappers to resolve. */
```

Comment:
I'll rework the comment, to say this is expect, but would like to keep something as its pretty confusing.